sklearn/svm/base.py

```diff
@@ -604,30 +603,31 @@ def _get_liblinear_solver_type(multi_class, penalty, loss, dual):
       - loss
       - dual
 
-    The same number is internally by LibLinear to determine which
-    solver to use.
+    The same number is also internally used by LibLinear to determine
+    which solver to use.
     """
-
-    # nested dicts containing level 1: available loss functions, 
+    if loss.isupper() or penalty.isupper():
+        warnings.warn(("Case insensitivity support for loss and penalty"
+                       " parameters is deprecated from 0.16 and will be"
+                       " removed in 0.18. The values for loss and"
+                       " penalty are expected to be in lower case only."),
```

Comment:
We need to provide explicit and informative warning and error messages that recall the values of both the observed and expected values. In this case:

```python
if loss in ('L1', 'L2'):
    warnings.warn("loss='%s' has been deprecated in favor of loss='%s' as of 0.16. "
               "Backward compatibility for the uppercase notation will be removed in 0.18"
                % (loss, loss.lower()))
```
and a similar message for `penalty`.

We should not assume that the user has the source code at hand with the precise line where this class is instantiated when executing a program that causes this message to be raised. Putting explicit parameter values in the message is likely to help him / her find the source of the problem much more quickly.